expression/builtin_cast_vec.go

```diff
@@ -221,7 +221,7 @@ func (b *builtinCastRealAsStringSig) vecEvalString(input *chunk.Chunk, result *c
 			result.AppendNull()
 			continue
 		}
-		res, err = types.ProduceStrWithSpecifiedTp(strconv.FormatFloat(v, 'f', -1, bits), b.tp, sc, false)
+		res, err = types.ProduceStrWithSpecifiedTp(strconv.FormatFloat(v, 'e', -1, bits), b.tp, sc, false)
```

Comment:
Looks like this change may cause compatible issues.

    drop table if exists t;
    create table t(c1 float(16, 4));
    insert into t1 values(3.1415926535);

Before change, compatible:

    mysql> select concat('a', c1) from t1;
    +-----------------+
    | concat('a', c1) |
    +-----------------+
    | a3.1416         |
    +-----------------+
    1 row in set (0.00 sec)

After changed, not compatible.

    mysql> select concat('a', c1) from t1;
    +-----------------+
    | concat('a', c1) |
    +-----------------+
    | a3.1416e+00     |
    +-----------------+